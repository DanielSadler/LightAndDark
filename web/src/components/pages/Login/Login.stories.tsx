import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Login } from './Login';

export default {
  title: 'Pages/Login',
  component: Login,
} as ComponentMeta<typeof Login>;

const Template: ComponentStory<typeof Login> = (args) => <Login {...args} />;

export const Primary = Template.bind({});
Primary.args = {};
