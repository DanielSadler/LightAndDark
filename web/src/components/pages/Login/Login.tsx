import Image from 'next/image';
import React, { FC } from 'react';
import { LoginForm } from '../../molecules';

type Props = {};

export const Login: FC<Props> = () => (
  <div className="min-h-full flex flex-col justify-center py-12 sm:px-6 lg:px-8">
    <div className="sm:mx-auto sm:w-full sm:max-w-md">
      <div className="flex justify-center">
        <Image width="283" height="64" src="/vercel.svg" alt="Logo" />
      </div>
      <h2 className="mt-6 text-center text-2xl font-bold text-gray-900">
        Sign in to your account
      </h2>
    </div>

    <LoginForm />
  </div>
);
