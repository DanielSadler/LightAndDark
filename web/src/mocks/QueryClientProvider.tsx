import React from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';

interface Props {
  children: React.ReactNode;
}

const client = new QueryClient();

export const Wrapper = ({ children }: Props) => (
  <QueryClientProvider client={client}>{children}</QueryClientProvider>
);
