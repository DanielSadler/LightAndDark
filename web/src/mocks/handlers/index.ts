import { exampleQuery } from './Example';

export const handlers = [exampleQuery];
