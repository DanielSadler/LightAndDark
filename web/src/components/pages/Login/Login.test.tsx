import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { expect } from '@jest/globals';
import { Login } from './Login';
import { Wrapper } from '../../../mocks/QueryClientProvider';

describe('Login', () => {
  it('should render', () => {
    render(
      <Wrapper>
        <Login />
      </Wrapper>
    );
  });
  it('should match a snapshot', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Login />
        </Wrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
