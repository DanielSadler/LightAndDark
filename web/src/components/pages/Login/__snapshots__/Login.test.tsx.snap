// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match a snapshot 1`] = `
<div
  className="min-h-full flex flex-col justify-center py-12 sm:px-6 lg:px-8"
>
  <div
    className="sm:mx-auto sm:w-full sm:max-w-md"
  >
    <div
      className="flex justify-center"
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27283%27%20height=%2764%27/%3e"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="Logo"
          data-nimg="intrinsic"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="Logo"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/vercel.svg"
            srcSet="/vercel.svg 1x, /vercel.svg 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
    </div>
    <h2
      className="mt-6 text-center text-2xl font-bold text-gray-900"
    >
      Sign in to your account
    </h2>
  </div>
  <div
    className="mt-8 sm:mx-auto sm:w-full sm:max-w-md"
  >
    <div
      className="bg-white py-8 px-4 shadow sm:rounded-lg sm:px-10"
    >
      <div
        className="space-y-6"
      >
        <div>
          <label
            className="block text-sm font-medium text-gray-700"
            htmlFor="email"
          >
            Email address
          </label>
          <div
            className="mt-1"
          >
            <input
              autoComplete="email"
              className="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
              id="email"
              name="email"
              onChange={[Function]}
              required={true}
              type="email"
              value=""
            />
          </div>
        </div>
        <div>
          <label
            className="block text-sm font-medium text-gray-700"
            htmlFor="email"
          >
            Password
          </label>
          <div
            className="mt-1"
          >
            <input
              autoComplete="current-password"
              className="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
              id="password"
              name="password"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          className="flex items-center justify-between"
        >
          <div
            className="flex items-center"
          >
            <input
              className="h-4 w-4 text-indigo-600 focus:ring-indigo-500 border-gray-300 rounded"
              data-testid="remember-me-checkbox"
              defaultChecked={false}
              id="remember-me"
              name="remember-me"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="ml-2 block text-sm text-gray-900"
              data-testid="remember-me-checkbox-label"
              htmlFor="remember-me"
            >
              Remember me
            </label>
          </div>
          <div
            className="text-sm"
          >
            <a
              className="font-medium text-indigo-600 hover:text-indigo-500"
              href="#"
            >
              Forgot your password?
            </a>
          </div>
        </div>
        <div>
          <button
            className="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
            onClick={[Function]}
            type="submit"
          >
            Sign in
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
